### PR TITLE
CGE2: Fix Sfinx crash in builds without TTS

### PR DESCRIPTION
**Pull request by PushmePullyu**
Add nullptr check before using TTS in the exit dialog.

Fixes [#14383](https://bugs.scummvm.org/ticket/14383)

Also make sure TTS only plays in the exit dialog when enabled in the game
options ("Enable Text to Speech for Objects and Options").

### CODE REVIEW

File: engines/cge2/vmenu.cpp [hunk 1]
```diff
@@ -155,7 +155,8 @@ void VMenu::touch(uint16 mask, V2D pos, Common::KeyCode keyCode) {
 
 		Common::TextToSpeechManager *ttsMan = g_system->getTextToSpeechManager();
 		if (_lastN != n) {
-			ttsMan->say(_menu[n]->_text, Common::TextToSpeechManager::INTERRUPT);
+			if (ttsMan != nullptr && ConfMan.getBool("tts_enabled_objects"))
```
Comment (lephilousophe) [inline engines/cge2/vmenu.cpp:158]: You also added a check without explaining why.

Comment (PushmePullyu) [reply engines/cge2/vmenu.cpp:158]: The game has two options for TTS:
1. ConfMan.getBool("tts_enabled_objects") for options and object descriptions
2. ConfMan.getBool("tts_enabled_speech") for subtitles

I think the quit dialog fits more into the first category.

Comment (lephilousophe) [reply engines/cge2/vmenu.cpp:158]: I was a little too terse.
I meant you should add a sentence in your commit message to explain that you added a missing check for settings.
Thanks

Comment (PushmePullyu) [reply engines/cge2/vmenu.cpp:158]: Updated. Hope it's not too verbose for a one line change.

**Comment by lephilousophe** [comment]
Perfect! Thank you.

**Pull request merged**